x-pack/plugins/files/public/components/common/image_metadata.ts

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import * as bh from 'blurhash';
+import type { FileImageMetadata } from '../../../common';
+
+export function isImage(file: Blob | File): boolean {
+  return file.type?.startsWith('image/');
+}
+
+export const boxDimensions = {
+  width: 300,
+  height: 300,
+};
+
+/**
+ * Calculate the size of an image, fitting to our limits see {@link boxDimensions},
+ * while preserving the aspect ratio.
+ */
+export function fitToBox(width: number, height: number): { width: number; height: number } {
+  const offsetRatio = Math.abs(
+    Math.min(
+      // Find the aspect at which our box is smallest, if less than 1, it means we exceed the limits
+      Math.min(boxDimensions.width / width, boxDimensions.height / height),
+      // Values greater than 1 are within our limits
+      1
+    ) - 1 // Get the percentage we are exceeding. E.g., 0.3 - 1 = -0.7 means the image needs to shrink by 70% to fit
+  );
+  return {
+    width: width - offsetRatio * width,
+    height: height - offsetRatio * height,
+  };
+}
+
+/**
+ * Get the native size of the image
+ */
+function loadImage(src: string): Promise<HTMLImageElement> {
+  return new Promise((res) => {
+    const image = new window.Image();
+    image.src = src;
+    image.onload = () => res(image);
+  });
+}
+
+/**
+ * Extract image metadata, assumes that file or blob as an image!
+ */
+export async function getImageMetadata(file: File | Blob): Promise<undefined | FileImageMetadata> {
```

Comment:
nit: `loadImage` and `getImageMetadata` are placed in `/common` folder, but they depend on `window` global, which will not work in Node.js.

Reply:
This is "common" but inside of the "public" folder. I can see how this clashes with other naming conventions. Will rename.